### PR TITLE
Feature/get internal dcm state

### PR DESCRIPTION
**Pull request by MiladShafiee**
We open this pull request for adding the feature of feeding some desired values(DCMSubTrajectories) from unicycle planner to step adjustment.

### CODE REVIEW

**Comment by S-Dafarra** [comment]
Some files we marked as modified because of some whitespaces. I removed them.

File: include/UnicycleGenerator.h [hunk 1]
```diff
@@ -53,7 +53,7 @@ class UnicycleGenerator {
 
     bool setTerminalHalfSwitchTime(double lastHalfSwitchTime); //if not set, it won't bring the ZMP at the center of the feet at the end
 
-    bool setPauseConditions(double maxStepTime, double nominalStepTime);
+    bool setPauseConditions(double maxStepTime, double nominalStepTime, bool pauseActive = false);
```
Comment (S-Dafarra) [inline include/UnicycleGenerator.h]: To be honest I don't like these for two reasons:
- It is ``false`` by default, meaning that it changes the behavior of this piece of code without being visible. What if some other code used this function before? After updating the behavior would be different without any apparent modification.
- It is not very clear when reading the code. Also, why would you set the pause conditions and deactivate them in the same method?

I would suggest adding a method like
```c++
void disablePauseConditions()
```
to disable pause conditions. On the contrary, the method ``setPauseConditions`` should enable them automatically.

Comment (MiladShafiee) [reply include/UnicycleGenerator.h]: > I would suggest adding a method like
> void disablePauseConditions()

This modification done in this commit. https://github.com/robotology/unicycle-footstep-planner/pull/34/commits/e5704c2d2c7652b793751e9aa2c4be6f53e5957c

> to disable pause conditions. On the contrary, the method setPauseConditions should enable them automatically.

Has it already been implemented ?

Comment (S-Dafarra) [reply include/UnicycleGenerator.h]: > Has it already been implemented ?

I think so: https://github.com/robotology/unicycle-footstep-planner/blob/fd10aae80e244feeba8f8c669181d1cad4e7f9a2/src/UnicycleGenerator.cpp#L590

File: include/FootPrint.h [hunk 2]
```diff
@@ -50,6 +48,8 @@ class FootPrint {
 
     const StepList& getSteps() const;
 
+    StepList& getSteps();
```
Comment (S-Dafarra) [inline include/FootPrint.h]: An additional question, why do you need this? Do you need to modify the list of steps?

Comment (MiladShafiee) [reply include/FootPrint.h]: Yes we update this the step positions and timing after step adaptation for re-planning the DCM trajectory, If I remember correctly.

Comment (S-Dafarra) [reply include/FootPrint.h]: Ok I see!

**Comment by MiladShafiee** [comment]
`Please @MiladShafiee try to understand why (in devel is not failing).`

I'm just wondering why when I run this test in Debug mode it fails, however, when I run it in the Release mode it runs without error!

**Comment by S-Dafarra** [comment]
This may be due to some ``assert`` that disappear when compiling in Release mode.

**Comment by MiladShafiee** [comment]
> The `DcmInterpolationTest` is failing with the message
> 
> > First run: start time 0 seconds
> > [DCMTrajectoryGeneratorHelper::evaluateDCMTrajectory] Error when the position of the ZMP is evaluated.
> > [DCMTrajectoryGeneratorHelper::generateDCMTrajectory] Error when the whole DCM trajectory is evaluated.
> > [DCMTrajectoryGenerator::interpolateDCM] Failed while computing the DCM trajectories.
> > [UnicycleGenerator::generate] Failed while computing new DCM trajectories.
> > :-1 : assertTrue failure
> 
> Please @MiladShafiee try to understand why (in `devel` is not failing).

The bug has been removed and I think now the test works properly.
I think we can merge it.

**Comment by S-Dafarra** [comment]
Thanks @MiladShafiee!

**Pull request merged**